ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +268,155 @@ void PowerSystemNode::injectFault (const string& fault_name,
     {
       ros::param::getCached("/faults/" + FAULT_NAME_HPD, hpd_wattage);
       wattage += hpd_wattage;
-    } else
+    }
+  }
+}
+
+void PowerSystemNode::injectCustomFault(bool& fault_activated,
+                                        const PrognoserVector& sequence,
+                                        size_t& index,
+                                        double& wattage,
+                                        double& voltage,
+                                        double& temperature)
+{
+  static string saved_fault_directory = "N/A";
+  string current_fault_directory;
+  string designated_file;
+  static string saved_file;
+  static bool custom_fault_ready = false;
+  static bool custom_warning_displayed = false;
+  static bool end_fault_warning_displayed = false;
+  bool fault_enabled = false;
+
+  // Do nothing unless the specified fault has been injected.
+  if (! ros::param::getCached("/faults/activate_custom_fault", fault_enabled)) {
+    return;
+  }
+
+  if (!fault_activated && fault_enabled)
+  {
+    // Only continue if no other faults have been injected.
+    ROS_INFO_STREAM("Attempting custom fault activation...");
+    if (HPD_FAULT_ACTIVATED)
+    {
+      if (!custom_warning_displayed)
+      {
+        ROS_WARN_STREAM("Cannot activate custom fault: high power draw already used this simulation.");
+        custom_warning_displayed = true;
```

Comment:
This warning is no longer valid, and yet this code seems unreachable.  At least I don't get this warning when I perform what it says it cannot do.

Reply:
Yup, it was deprecated as of commenting out the flags that were set upon activating either a custom fault or high power draw. I've removed them in the incoming commit (along with all of the other changes you've requested).